### PR TITLE
Bump GKE to 1.19

### LINKED ISSUE #1170: Bump GKE version to 1.19
Bump GKE cluster version and node pool version to 1.19.  During upgrade, if the cluster is created then trigger the upgrade to 1.19

### PR DESCRIPTION
**Pull request by sreis**
Close #1170

### CODE REVIEW

**Comment by sreis** [comment]
Yesterday I wrote a bunch of code to upgrade the control plane and the node pools but today while testing the GKE upgrade locally I noticed that by default the `autoUpgrade` feature is enabled. 

Snip of [`existingGKEcluster struct`](https://github.com/opstrace/opstrace/blob/main/lib/gcp/src/cluster.ts#L184):
```
      "management": {
        "autoUpgrade": true,
        "autoRepair": true,
        "upgradeOptions": null
      },

```

This is nice! It means we can probably just bump the initial cluster version and GKE will take care of the rest for us.

We have a couple of long-running clusters in GKE that have been upgraded regularly and we haven't had any issues.

**Comment by sreis** [comment]
Following tests failed in the GCP test that creates the cluster with Kubernetes version 1.19. They also failed in the previous CI build so I need to investigate these further.

```
[2021-08-04T11:50:19Z]   1) End-to-end alert tests
[2021-08-04T11:50:19Z]        End-to-end alerts for default and system tenants:
[2021-08-04T11:50:19Z]      Error: kubectl port-forward setup failed
[2021-08-04T11:50:19Z]       at PortForward.setup (testutils/portforward.ts:245:11)
[2021-08-04T11:50:19Z]       at processTicksAndRejections (node:internal/process/task_queues:96:5)
[2021-08-04T11:50:19Z]       at Object.waitForPrometheusTarget (testutils/metrics.ts:71:21)
[2021-08-04T11:50:19Z]       at waitForE2EAlertFiring (test_alerts.ts:388:3)
[2021-08-04T11:50:19Z]       at Context.<anonymous> (test_alerts.ts:453:5)
[2021-08-04T11:50:19Z] 
[2021-08-04T11:50:19Z]   2) system logs test suite
[2021-08-04T11:50:19Z]        check loki ingester logs:
[2021-08-04T11:50:19Z]      Error: Expectation not fulfilled within 30 s
[2021-08-04T11:50:19Z]       at Object.waitForLokiQueryResult (testutils/logs.ts:265:9)
[2021-08-04T11:50:19Z]       at query (test_systemlogs.ts:52:20)
[2021-08-04T11:50:19Z]       at Context.<anonymous> (test_systemlogs.ts:66:5)
[2021-08-04T11:50:19Z] 
[2021-08-04T11:50:19Z]   3) system logs test suite
[2021-08-04T11:50:19Z]        check cortex distributor logs:
[2021-08-04T11:50:19Z]      Error: Expectation not fulfilled within 30 s
[2021-08-04T11:50:19Z]       at Object.waitForLokiQueryResult (testutils/logs.ts:265:9)
[2021-08-04T11:50:19Z]       at query (test_systemlogs.ts:52:20)
[2021-08-04T11:50:19Z]       at Context.<anonymous> (test_systemlogs.ts:72:5)
[2021-08-04T11:50:19Z] 
[2021-08-04T11:50:19Z]   4) system logs test suite
[2021-08-04T11:50:19Z]        check systemlog Fluentd instance logs:
[2021-08-04T11:50:19Z]      Error: Expectation not fulfilled within 30 s
[2021-08-04T11:50:19Z]       at Object.waitForLokiQueryResult (testutils/logs.ts:265:9)
[2021-08-04T11:50:19Z]       at query (test_systemlogs.ts:52:20)
[2021-08-04T11:50:19Z]       at Context.<anonymous> (test_systemlogs.ts:78:5)
[2021-08-04T11:50:19Z] 
```

```
[2021-08-04T11:52:55Z]   1) test_ui_api
[2021-08-04T11:52:55Z]        test_grafana_datasource_proxy_loki_get_labels:
[2021-08-04T11:52:55Z]      Error: unexpected response
[2021-08-04T11:52:55Z]       at Context.<anonymous> (test_ui_api.ts:316:11)
[2021-08-04T11:52:55Z]       at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

**Comment by sreis** [comment]
Couldn't repro the `Error: kubectl port-forward setup failed` error locally

```
PortForward(tenant-prometheus-default): exec in child: 'kubectl port-forward --namespace default-tenant statefulsets/prometheus-default-prometheus 0:9090' -- redirect output to /tmp/opstrace-test-remote/kubectl-port-forward--29-JM3JVU5Dq6C7-.outerr
```

I ssh'ed to the buildkice CI instance and this was in the file

```
[ec2-user@ip-10-0-2-116 opstrace-test-remote]$ sudo cat /tmp/opstrace-test-remote/kubectl-port-forward--29-JM3JVU5Dq6C7-.outerr
I0804 11:46:32.498662      40 request.go:621] Throttling request took 1.103824288s, request: GET:https://34.94.72.236/apis/networking.gke.io/v1?timeout=32s
Forwarding from 127.0.0.1:33933 -> 9090
Forwarding from [::1]:33933 -> 9090
```

The failing test reads 100 bytes and then uses a regex to check the port-forwarding is set up. I bumped to 500 bytes and updated the regex.

**Comment by sreis** [comment]
Tracking down why queries are failing with 429s (too many outstanding requests) and also why it can't start the  grafana websocket connection.

Screenshot from 2021-08-05 14-53-40

```
> curl --silent -m 30 -H "Authorization: Bearer $(cat tenant-api-token-system)" -k "https://cortex.system.sreis.opstrace.io/api/v1/labels"  
too many outstanding requests%  
```

**Comment by sreis** [comment]
Diff'ed the Cortex config from an Opstrace instance I created locally with the long-running collection instance and didn't find any differences besides new default settings.

Then, I restarted the query-frontend and it seems to have fixed the cortex querying issues... which is very odd.

Next up are the Loki issues. They seem to be related to fluentd being unable to parse the system logs. Fluentd is also OOM'ing after running for a few minutes.

```
2021-08-05 15:43:09 +0000 [warn]: #0 pattern not matched: "2021-08-05T15:43:06.255321724Z stdout P 2021-08-05 15:43:06 +0000 [warn]: #0 pattern not matched: \"2021-08-05T15:43:04.445954609Z stdout F 2021-08-05 15:43:04 +0000 [warn]: #0 pattern not matched: \\\"2021-08-05T15:43:03.156482056Z stdout F 2021-08-05 15:43:03 +0000 [warn]: #0 pattern not matched: \\\\\\\"2021-08-05T15:43:02.054866379Z stdout F 2021-08-05 15:43:02 +0000 [warn]: #0 pattern not matched: \\\\\\\\\\\\\\\"2021-08-05T15:43:00.967600448Z stdout F 2021-08-05 15:43:00 +0000 [warn]: #0 pattern not matched: \\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\"2021-08-05T15:42:59.966388624Z stdout F 2021-08-05 15:42:59 +0000 [warn]: #0 pattern not matched: \\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\"2021-08-05T15:42:58.957486916Z stdout F 2021-08-05 15:42:58 +0000 [warn]: #0 pattern not matched: \\\\\\\\
```

**Comment by sreis** [comment]
Fixed the fluentd issues locally by installing the cri gem and configuring fluentd to use it on GKE. Let's see if we need to force an upgrade to 1.18 or not.

**Comment by sreis** [comment]
Upgrade tests are green which is very nice so I'll go ahead and merge.

File: containers/systemlog-fluentd/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM fluent/fluentd:v1.12.0-debian-1.0
+FROM fluent/fluentd:v1.12.4-debian-1.0
```
Comment (jgehrcke) [inline containers/systemlog-fluentd/Dockerfile:1]: thanks!

File: packages/controller/src/resources/apis/systemlogs.ts [hunk 2]
```diff
@@ -43,6 +43,17 @@ export function SystemLogAgentResources(
...
 
+  const { target } = getControllerConfig(state);
+
+  // We need to use the CRI parser on GKE 1.19+
+  let parser = "";
+  if (target === "gcp") {
+    parser = `  <parse>
+    @type cri
+    merge_cri_fields false
+  </parse>`;
```
Comment (jgehrcke) [inline packages/controller/src/resources/apis/systemlogs.ts:54]: :rocket:

**Pull request merged**